oauth2_http/javatests/com/google/auth/oauth2/CertificateIdentityPoolSubjectTokenSupplierTest.java

```diff
@@ -143,4 +157,274 @@ public void getSubjectToken_success() throws Exception {
+
+    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+    // Parse and encode the leaf certificate.
+    X509Certificate expectedLeafCert =
+        (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(testCertBytesFromFile));
+    String expectedEncodedLeaf = Base64.getEncoder().encodeToString(expectedLeafCert.getEncoded());
+
+    // Parse and encode the intermediate certificate.
+    X509Certificate expectedIntermediateCert =
+        (X509Certificate)
+            cf.generateCertificate(new ByteArrayInputStream(intermediateCertBytesFromFile));
+    String expectedEncodedIntermediate =
+        Base64.getEncoder().encodeToString(expectedIntermediateCert.getEncoded());
+
+    // Expected: JSON array with encoded leaf and then encoded intermediate.
+    String[] expectedCertChain = new String[] {expectedEncodedLeaf, expectedEncodedIntermediate};
+    String expectedSubjectToken = OAuth2Utils.JSON_FACTORY.toString(expectedCertChain);
+
+    // Execute
+    String actualSubjectToken = supplier.getSubjectToken(mockContext);
+
+    // Verify
+    assertEquals(expectedSubjectToken, actualSubjectToken);
+  }
+
+  @Test
+  public void getSubjectToken_trustChainWithoutLeaf_success() throws Exception {
+    // Configure mock to return the path to the trust chain file WITHOUT leaf.
+    ClassLoader classLoader = getClass().getClassLoader();
+    URL trustChainUrl = classLoader.getResource("trust_chain_without_leaf.pem");
+    assertNotNull("Test trust chain file (without leaf) not found!", trustChainUrl);
+    when(mockCertificateConfig.getTrustChainPath())
+        .thenReturn(new File(trustChainUrl.getFile()).getAbsolutePath());
+
+    CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+    // Parse and encode the leaf certificate.
+    X509Certificate expectedLeafCert =
+        (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(testCertBytesFromFile));
+    String expectedEncodedLeaf = Base64.getEncoder().encodeToString(expectedLeafCert.getEncoded());
+
+    // Parse and encode the intermediate certificate.
+    X509Certificate expectedIntermediateCert =
+        (X509Certificate)
+            cf.generateCertificate(new ByteArrayInputStream(intermediateCertBytesFromFile));
+    String expectedEncodedIntermediate =
+        Base64.getEncoder().encodeToString(expectedIntermediateCert.getEncoded());
+
+    // Expected: JSON array with encoded leaf and then encoded intermediate.
+    String[] expectedCertChainArray =
+        new String[] {expectedEncodedLeaf, expectedEncodedIntermediate};
+    String expectedSubjectToken = OAuth2Utils.JSON_FACTORY.toString(expectedCertChainArray);
+
+    // Execute
+    String actualSubjectToken = supplier.getSubjectToken(mockContext);
+
+    // Verify
+    assertEquals(expectedSubjectToken, actualSubjectToken);
+  }
+
+  @Test
+  public void getSubjectToken_trustChainWrongOrder_throwsIllegalArgumentException() {
```

Comment:
Can you add a comment above this test to clarify what wrong order means? I believe this case would be that the leaf is not first